source/common/http/url_utility.cc

```diff
@@ -0,0 +1,124 @@
+#include "common/http/url_utility.h"
+
+#include "common/common/assert.h"
+#include "common/common/empty_string.h"
+
+#include "absl/strings/str_cat.h"
+
+#ifndef WIN32
+#include "url/gurl.h"
+#else
+#include <http_parser.h>
+#endif
+
+namespace Envoy {
+namespace Http {
+namespace Utility {
+
+#ifndef WIN32
+constexpr char COLON[] = ":";
+constexpr char HASH[] = "#";
+#else
+constexpr char DEFAULT_PATH[] = "/";
+constexpr char HTTP[] = "http";
+constexpr char HTTPS[] = "https";
+#endif
+
+bool Url::initialize(absl::string_view absolute_url) {
```

Comment:
At some point, can we get rid of https://github.com/envoyproxy/envoy/tree/master/source/common/chromium_url by doing the normalization in this class via GURL?

Reply:
I think so yeah.